lib/zipline/zip_generator.rb

```diff
@@ -44,9 +44,7 @@ def normalize(file)
         {file: File.open(file.path)}
       elsif is_io?(file)
         {file: file}
-      elsif defined?(ActiveStorage::Blob) && file.is_a?(ActiveStorage::Blob)
-        {url: file.service_url}
-      elsif defined?(ActiveStorage::Attachment) && file.is_a?(ActiveStorage::Attachment)
+      elsif defined?(ActiveStorage::Blob) && file.respond_to?(:blob) && file.blob.is_a?(ActiveStorage::Blob)
         {url: file.blob.service_url}
```

Comment:
if the file IS a blob, we don't want to call file.blob.service_url, just file.service_url

Reply:
Done. Also added test